accepted/00070-akka.md

```diff
@@ -0,0 +1,134 @@
+
+With reference to the open problems stated above:
+
+1. in Akka dependencies can be expressed by actors firing messages to other actors when done
+2. any other part of the code base can ask an Akka actor to do something by sending it a message
+3. [Akka has a concept of priority messages to actors](https://doc.akka.io/docs/akka/2.5/mailboxes.html)
+4. Akka seems to be regarded as high performance, has thorough and well-documented tuning knobs
+5. [Akka has a lot of utilities for monitoring](https://developer.lightbend.com/docs/telemetry/current/home.html)
+6. Akka is generally known and well documented, has books, videos and other material available
+
+## Detail design ideas
+
+ - The Akka-based asynchronous facility will be made available via a facade (`ActorManager` in the above PR)
+   - any code needing to run an asynchronous computation will use the facade
+   - the facade will accept messages to Actors (`Command`s), instantiated by calling code
+   - the facade will route `Command`s to respective `Actor`s - `Actor`s normally receiving one `Command` type and corresponding to one computation unit (current `Action`s, `Task`s and so on)
+   - all `Commands` should be `Serializable` and contain simple data types as much as possible
+   - the "functional" Akka style is preferred when writing Actors
+
+## Steps
+
+1. package Akka and integrate it with the Uyuni product packages, build systems
+2. move Tomcat's `MessageQueue`-based Actions to Akka
+3. move `PGEventStream`-based Salt event handlers to Akka
+4. drop `MessageQueue` and related mechanisms
+
+All of the above was implemented (as proof-of-concept) as part of [this HackWeek PR](https://github.com/uyuni-project/uyuni/pull/1903)  (excluding commits after February 15, which deal with Akka Clustering) which can serve as a reference for implementation details.
+
+Some shortcuts have been taken during HackWeek and will not be acceptable for proper implementation, particularly:
+
+ - unit tests have been arbitrarily disabled or left broken
+ - new methods/classes do not have new accompanying unit tests
+ - logging is incomplete
+ - code style of Akka-related parts follows the Akka coding conventions, which are different from the CheckStyle rules currently applied. We will either need to adapt new code to the existing conventions or relax them/create different rules
+ - error handling was not tested thoroughly, might need rework
+ - the Cucumber integration testsuite was not run on the resulting code
+ - performance tests will have to be conducted to make sure there is no regression (and evaluate possible speedups)
+ - the reflections-based mechanism to find out workers will have to be considered and it might be removed in exchange for an explicit approach
+ - most but not all Actor Messages were made `Serializable`, more refactoring might be needed to ensure this is always the case
```

Comment:
also no exhaustive testing in the refactorings we have done for messages when transforming messages to be serializable. I remember some classes we move some Id's to the message but the old code was doing some processing with it, and we replace it by computing the data again. I Don't know the impact of this behavior change.